docs/configurations/proxy.md

```diff
@@ -1,120 +1,225 @@
         {
-            "type": "text",
-            "label": "Password",
-            "validators": [
-                {
-                    "type": "string",
-                    "errorMsg": "Max length of password is 8192",
-                    "minLength": 0,
-                    "maxLength": 8192
-                }
-            ],
-            "encrypted": true,
-            "field": "proxy_password"
-        },
+            "type": "string",
+            "errorMsg": "Max length of username is 50",
+            "minLength": 0,
+            "maxLength": 50,
+        }
+    ],
+    "field": "proxy_username",
+}
+```
+
+- Password :
+
+```json
+{
+    "type": "text",
+    "label": "Password",
+    "validators": [
         {
-            "type": "checkbox",
-            "label": "Reverse DNS resolution",
-            "field": "proxy_rdns"
+            "type": "string",
+            "errorMsg": "Max length of password is 8192",
+            "minLength": 0,
+            "maxLength": 8192,
         }
     ],
-    "options": {
-        "saveValidator": "function(formData) { if(!formData.proxy_enabled || formData.proxy_enabled === '0') {return true; } if(!formData.proxy_url) { return 'Proxy Host can not be empty'; } if(!formData.proxy_port) { return 'Proxy Port can not be empty'; } return true; }"
+    "encrypted": true,
+    "field": "proxy_password",
+}
+```
+
+- DNS Resolution checkbox :
+
+```json
+{
+    "type": "checkbox",
+    "label": "DNS resolution",
+    "field": "proxy_rdns",
+}
+```
+
+If you only specify `"type": "proxyTab"` in your globalConfig file, then entities such as `proxy_type`, `username`, `password`, and `dns_resolution` will not be generated, only `host`, `port`, `enabled` entities will be generated.  To include the optional entities in your add-on (refer [available configurations](#available-configurations) listed above), you need to set them to `True`.
+
+### Usage
+
+It is placed just like every other configuration tab.
+
+```json
+{
+    "pages": {
+        "configuration": {
+            "tabs": [
+              {
+                "type": "proxyTab"
+              }
+            ],
+            "title": "Configuration",
+            "description": "..."
+        }
     }
 }
 ```
 
-The example above creates the following form:
+TTo customize these entities, you can define them in JSON format by specifying certain keys unique to each entity (refer the [keys section](#keys) for details on the required keys for each entity). When creating the JSON, you only need to include the values you want to customize. For example, if you don’t need to modify fields like `lable` or `field`, you can skip including them in your globalConfig.  For better understanding, refer to the provided [Example](#example) which demonstrates how to apply customizations effectively.
+
+> **_NOTE:_**
+    There are 2 ways to exclude optional entities in your add-on, either omit them from the proxy tab, or set the entities to false.
+
+### Keys
```

Comment:
Now I think: wouldn't it be good to put this table above `### Available configurations`? Plus add "name" and "title" to the table and indicate required and optional fields.

Reply:
I think moving this table above `Available configuration` will break the flow of the document, as first we should mention what is being generated by default and then steps on how to customize it, which we are following here. 
I have added `name` and `title` keys in the Keys table.

Reply:
Ok. So maybe would it be good to indicate what the `### Available configurations` is about? To mention something about those values.

E.g.:

````
- Enable proxy :

```json
{
    "type": "checkbox", 
    "label": "Enable",
    "field": "proxy_enabled"
}
```
````

Because when I had a first look at that, I wasn't sure whether this is a parameter to set, or does it say that it is the default value for the minimal definition.

Reply:
I had made the changes in documentation, can you please look into it?